.github/workflows/build-push-via-cicd-actions.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Push Image #by triggering cicd-actions workflow via Github API Call
+
+on:
+  push:
+    branches: [ feature/custatevec, feature/ZCLOUD-260/add-ci-with-github-actions ]
+  
+jobs:
+  trigger-build-and-push:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Trigger cicd-actions repository workflow over Github API
+        run: |
+              curl \
+              "https://api.github.com/repos/zapatacomputing/cicd-actions/dispatches" \
+              -H "Authorization: token "$USER_TOKEN \
+              -H 'Accept: application/vnd.github.everest-preview+json' \
+              --data-raw '
+              {
+                "event_type":  "'"$repository"' | '"$ref"'", 
+                "client_payload": 
+                    {
+                      "repository": "'"$repository"'", 
+                      "ref": "'"$ref"'", 
+                      "path_to_dockerfile": "'"$path_to_dockerfile"'", 
+                      "docker_context_path": "'"$docker_context_path"'", 
+                      "image_version": "'"$image_version"'" 
+                    } 
+                }
+              '
+        env:
+          USER_TOKEN: ${{ secrets.cicd_actions_user }}
+          repository: ${{ github.repository }}
+          ref: ${{ github.ref }}
+          image_version: v${{ github.run_number }}
```

Comment:
we may want image_version to be configurable, in one PR i may want to use `zapatacomputing/z-quantum-gpu:python38` etc. We can do this pretty easy just setting an env var in this file during a PR